terraform/mock-provider/container_image.tf

```diff
@@ -28,13 +28,34 @@ resource "null_resource" "mock-provider_image_push" {
   }
 
   provisioner "local-exec" {
-    command = <<EOF
+    interpreter = ["bash", "-c"]
+    command=<<-EOF
 export AWS_PROFILE=apim-dev
 aws ecr get-login-password --region eu-west-2 | docker login --username AWS --password-stdin ${data.aws_caller_identity.current.account_id}.dkr.ecr.eu-west-2.amazonaws.com
 ecr_url=${data.aws_ecr_repository.mock_provider_repository.repository_url}
 image_tag=$ecr_url:${var.image_version}
 docker build -t $image_tag -f ${local.mock_provider_path}/Dockerfile ${local.mock_provider_path}
 docker push -a $ecr_url
-       EOF
+aws ecs update-service --cluster ${var.prefix} --service ${var.prefix} --force-new-deployment --region eu-west-2
+sleep 50
+counter=0
+endpoint=https://${var.service_domain_zone}/_status
```

Comment:
This shouldn't be a var. It's comming from the `api` module. So it's an output from that module. something like: `module.api.outputs.service_domain_zone`, given you declare it as an output in that module.
Also since the value is in terraform, please remove all the occurances of this in makefile. There is no need for anyone to get that url outside the terraform.

Reply:
We cannot remove this output from Makefile, as we need this in azure script for populating KVM

Reply:
Also I have tried using `module.api.outputs.service_domain_zone`, directly but its giving error, we cannot directly use other module variable like this. Yeah i can use `outptut` command here again, should i go that way?

Reply:
What's the error? Make sure the module actually has an output with the same name. I just put that as an example. 
In terraform you have a value. In order to get/use that value you need to define it as an output in that module. Then in the caller module you can just use it by referencing it. It should work.

Reply:
Both `Mock-provider` and `API` are child modules of `terraform`, so we cant directly access output of `API` in `mock-provider`. We need to access them via main terraform directory, where both modules are defined. Thats why created variable, which is initialized when `mock-provider` module is defined